### PR TITLE
[20.03] openssh: 8.1p1 -> 8.2p1

### PR DESCRIPTION
**Pull request by prusnak**
###### Motivation for this change

Backport of OpenSSH 8.2p1 as discussed in https://github.com/NixOS/nixpkgs/pull/80196

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libfido2/default.nix [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
   };
 
   nativeBuildInputs = [ cmake pkgconfig ];
-  buildInputs = [ libcbor libressl udev ];
+  buildInputs = [ libcbor libressl ]
+    ++ stdenv.lib.optionals stdenv.isLinux [ udev ]
+    ++ stdenv.lib.optionals stdenv.isDarwin [ IOKit ];
 
-  cmakeFlags = [ "-DUDEV_RULES_DIR=${placeholder "out"}/etc/udev/rules.d" ];
+  patches = [ ./detect_apple_ld.patch ];
```
Comment (flokli) [inline pkgs/development/libraries/libfido2/default.nix]: can this patch be upstreamed and `fetchurl`-ed instead? We usually try to not ship patches inside nixpkgs.

Comment (prusnak) [reply pkgs/development/libraries/libfido2/default.nix]: Fixed in 10c787daddb2baceb60fe909971c233de4f0755b

**Comment by worldofpeace** [comment]
Aren't we going to have to forward port https://github.com/NixOS/nixpkgs/pull/81368/commits/10c787daddb2baceb60fe909971c233de4f0755b if we merge this? Is that on staging?

**Comment by prusnak** [comment]
@worldofpeace I don't think that's necessary. There will be a new version of libfido2 in April and I'll do a version update to unstable dropping the patch at that point. However, if you think it's worth changing in unstable too, I'll send a PR.

**Comment by worldofpeace** [comment]
@prusnak This probably should have been suggested on the initial PR https://github.com/NixOS/nixpkgs/pull/81368#discussion_r386080365. It's vanity divergence, so it would be best if it was like this on master? I believe it's still in staging.

**Comment by prusnak** [comment]
I dropped the 10c787d from this PR.

**Comment by worldofpeace** [comment]
Thanks @prusnak, I hope you're ok with this @flokli. This should just be a straightforward backport. No real reason to diverge things unless there's a good reason.

**Pull request merged**